pkg/quarterdeck/orgs.go

```diff
@@ -14,12 +14,92 @@ import (
 	"github.com/rotationalio/ensign/pkg/quarterdeck/permissions"
 	"github.com/rotationalio/ensign/pkg/quarterdeck/tokens"
 	"github.com/rotationalio/ensign/pkg/utils/metrics"
+	"github.com/rotationalio/ensign/pkg/utils/pagination"
 	"github.com/rotationalio/ensign/pkg/utils/sentry"
 	"github.com/rotationalio/ensign/pkg/utils/ulids"
 )
 
 const OneTimeAccessDuration = 5 * time.Minute
 
+func (s *Server) OrganizationList(c *gin.Context) {
+	var (
+		err                error
+		userID             ulid.ULID
+		orgs               []*models.Organization
+		nextPage, prevPage *pagination.Cursor
+		claims             *tokens.Claims
+		out                *api.OrganizationList
+	)
+
+	query := &api.OrganizationPageQuery{}
+	if err = c.BindQuery(query); err != nil {
+		sentry.Warn(c).Err(err).Msg("could not parse api page query")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse(api.ErrUnparsable))
+		return
+	}
+
+	if query.NextPageToken != "" {
+		if prevPage, err = pagination.Parse(query.NextPageToken); err != nil {
+			sentry.Warn(c).Err(err).Msg("could not parse next page token")
+			c.JSON(http.StatusBadRequest, api.ErrorResponse(err))
```

Comment:
Since we will have the error in Sentry, can we return a more generic error here (since it will be returned to the user)? (e.g. something like "invalid next page token")